src/main/java/com/nextcloud/common/NextcloudClient.kt

```diff
@@ -50,42 +47,58 @@ import javax.net.ssl.SSLSession
 import javax.net.ssl.TrustManager
 
 
-class NextcloudClient(var baseUri: Uri, val context: Context) {
+class NextcloudClient(var baseUri: Uri, val context: Context, client: OkHttpClient) {
     lateinit var credentials: String
     lateinit var userId: String
     lateinit var request: Request
     var followRedirects = true;
-    val client: OkHttpClient
+    val client = client
+
+    data class ResponseOrError private constructor(val result: Response?, val error: Exception?) {
```

Comment:
Nice approach :+1: 
To not have too many things in one class, can we extract this into its own class?

Reply:
extracted